consensus/replay.go

```diff
@@ -469,9 +469,23 @@ func (h *Handshaker) replayBlocks(
 			assertAppHashEqualsOneFromBlock(appHash, block)
 		}
 
-		appHash, err = sm.ExecCommitBlock(proxyApp.Consensus(), block, h.logger, h.stateStore, h.genDoc.InitialHeight)
-		if err != nil {
-			return nil, err
+		if i == finalBlock {
+			// We emit events for the index services at the final block due to the sync issue when
+			// the node shutdown during the block committing status.
```

Comment:
Do you think it's possible for the block to emit the same events twice?

Reply:
Yes, if `mutateState` flag is true. I will revise it. Do you think we need fireEvent if the flag is false? If no need I can rollback the change in this part

Reply:
I think its fine to leave this here if `mutateState` is false

Reply:
Cool. Thanks!